docs/DEVELOPING.md

```diff
@@ -343,6 +343,114 @@ $ make ftest NAME=azure_blob_storage
+
+The [Elastic Microsoft SQL connector](https://github.com/elastic/connectors-python/blob/8.7/connectors/sources/mssql.py) is provided in the Elastic connectors python framework and can be used via [build a connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
+
+### Availability and prerequisites
+
+⚠️ _Currently, this connector is available in **beta** starting in 8.7_.
+Features in beta are subject to change and are not covered by the service level agreement (SLA) of features that have reached general availability (GA).
+
+Elastic versions 8.6.0+ are compatible with Elastic connector frameworks. Your deployment must include the Elasticsearch, Kibana, and Enterprise Search services.
+
+**Prerequisite**
+
+The Microsoft SQL connector requires an ODBC driver to execute. To install ODBC driver please refer following [documentation](https://learn.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-driver-manager).
+
+### Setup and basic usage
+
+Complete the following steps to deploy the connector:
+
+1. [Gather Microsoft SQL details](#gather-microsoft-sql-details)
+2. [Configure Microsoft SQL connector](#configure-microsoft-sql-connector)
+
+#### Gather Microsoft SQL details
+
+Collect the information that is required to connect to your Microsoft SQL Server:
+
+- The server host address where the Microsoft SQL Server is hosted.
+- The port where the Microsoft SQL Server is hosted.
+- The username the connector will use to log in to the Microsoft SQL Server.
+- The password the connector will use to log in to the Microsoft SQL Server.
+- The database name where the connector will query data.
+
+#### Configure Microsoft SQL connector
+
+The following configuration fields need to be provided for setting up the connector:
+
+##### `host`
+
+The server host address where the Microsoft SQL Server is hosted. Default value is `127.0.0.1`. Examples:
+
+  - `192.158.1.38`
+  - `demo.instance.demo-region.demo.service.com`
+
+##### `port`
+
+The port where the Microsoft SQL Server is hosted. Default value is `9090`. Examples:
+
+  - `5432`
+  - `9090`
+
+##### `username`
+
+The username of the account for Microsoft SQL Server. Default value is `admin`.
+
+##### `password`
+
+The password of the account to be used for the Microsoft SQL Server. Default value is `Password_123`.
+
+##### `database`
+
+Name of the Microsoft SQL Server database. Default value is `xe`. Examples:
+
+  - `employee_database`
+  - `customer_database`
+
+##### `tables`
+
+Comma-separated list of tables. The Microsoft SQL connector will fetch data from all tables present in the configured database, if the value is `*` . Default value is `*`. Examples:
+
+  - `table_1, table_2`
+  - `*`
+
+##### `fetch_size`
+
+The number of rows to fetch on each request to Microsoft SQL Server. Default value is `50`.
+
+##### `retry_count`
+
+The number of retry attempts after failed request to Microsoft SQL Server. Default value is `3`.
```

Comment:
It can be helpful to provide some information on the tradeoff of increasing the # of retries and if there's a max # that can be retried or if if there's a setting for no retry limit.

Reply:
same as [this](https://github.com/elastic/connectors-python/pull/549#discussion_r1127228169) query.